src/contracts/examples/redstone-oev/RedstoneAdapterAtlasWrapper.sol

```diff
@@ -88,64 +100,87 @@ contract RedstoneAdapterAtlasWrapper is Ownable, MergedSinglePriceFeedAdapterWit
         uint256 minIntervalBetweenUpdates = getMinIntervalBetweenUpdates();
         if (currentBlockTimestamp < blockTimestampFromLatestUpdate + minIntervalBetweenUpdates) {
             revert MinIntervalBetweenUpdatesHasNotPassedYet(
-                currentBlockTimestamp,
-                blockTimestampFromLatestUpdate,
-                minIntervalBetweenUpdates
+                currentBlockTimestamp, blockTimestampFromLatestUpdate, minIntervalBetweenUpdates
             );
         }
     }
 
     function latestAnswer() public view virtual override returns (int256) {
         bytes32 dataFeedId = getDataFeedId();
-        uint256 baseAnswer = IAdapter(BASE_ADAPTER).getValueForDataFeed(dataFeedId);
+        (, uint256 atlasLatestBlockTimestamp) = getTimestampsFromLatestUpdate();
         uint256 atlasAnswer = getValueForDataFeed(dataFeedId);
-        if (atlasAnswer == 0) {
-            return int256(baseAnswer);
-        }
-        (uint256 baseLatestTimestamp,) = IAdapter(BASE_ADAPTER).getTimestampsFromLatestUpdate();
-        (uint256 atlasLatestTimestamp,) = getTimestampsFromLatestUpdate();
 
-        if (atlasLatestTimestamp > baseLatestTimestamp - BASE_FEED_DELAY) {
-            return int256(atlasAnswer);
+        uint80 roundId = IFeed(BASE_FEED).latestRound();
+
+        uint256 numIter;
+        for (;;) {
+            (uint256 baseValue,, uint128 baseBlockTimestamp) =
+                IAdapterWithRounds(BASE_ADAPTER).getRoundDataFromAdapter(dataFeedId, roundId);
+            if (atlasLatestBlockTimestamp >= baseBlockTimestamp) {
+                return int256(atlasAnswer);
+            }
+            if (block.timestamp - baseBlockTimestamp >= BASE_FEED_DELAY) {
+                return int256(baseValue);
+            }
+
+            unchecked {
+                numIter++;
+                roundId--;
+            }
+
+            if (numIter > MAX_HISTORICAL_FETCH_ITERATIONS) {
+                revert CannotFetchHistoricalData();
```

Comment:
Lets return the latestBaseAnswer in this worst case scenario.

Reply:
Done